### ISSUE TITLE
(Subform)Repeatable editor blank after drag&drop reorder

### ISSUE THREAD

**Issue by apophis1982** [joomla/joomla-cms#24343]
### Steps to reproduce the issue
Create a new Field with type repeatable and add under form fields one field type Editor. Set options to be shown in a tab in article edit view. Create a new article go to the tab and add 2 rows. Write something in each editor field and save and reorder the rows.

### Expected result
After reordering the rows editor behavior is normal

### Actual result
After you release the mouse button after reordering the rows the editor field gets blank and uneditable but the content will be saved and after refresh loaded correctly.

### System information (as much as possible)
Linux apache php 7.2 and 7.3. Joomla 3.9.4. Editor JCE Pro and TinyMCE (joomla standard). Also tried it with a new clean cloud installation via launch.joomla.

**Comment by ghost** [comment]
@HLeithner can you please comment?

**Comment by HLeithner** [comment]
I can confirm this problem, maybe someone with better js knowledge then me could look at this.

**Comment by AndySDH** [comment]
Repeatable field is going away in Joomla 4. It will be replaced by the better version, called Subfields
See: https://github.com/joomla/joomla-cms/pull/24711

Please test if the problem happens with this new version too and report in that PR.

**Comment by HLeithner** [comment]
We support 3.x for at least 2 years after j 4.0 release so this should be fixed.

**Comment by ghost** [comment]
closed for Reason stated above.

**Comment by ryandemmer** [comment]
The DOM is changed when the editors are sorted, so the editors need to be rebuilt.

Firstly, you need an event to be triggered when the sort is completed, which you can then use to trigger the re-initialisation of the editors.

As an example, an "update" option can be added to the sortable options in media/system/js/subform-repeatable.js to trigger a "subform-row-sort" :

See https://github.com/joomla/joomla-cms/blob/staging/media/system/js/subform-repeatable-uncompressed.js#L52-L58

```javascript
// bind move button
if (this.options.btMove) {
    this.$containerRows.sortable({
        items: this.options.repeatableElement,
        handle: this.options.btMove,
        tolerance: 'pointer',
        update: function (event, ui) {
            var $row = ui.item[0];
            self.$container.trigger('subform-row-sort', $row);
        }
    });
}
```
then in media/editors/tinymce/js/tinymce.js, a listener triggers the editor setup after the sort :
See - https://github.com/joomla/joomla-cms/blob/staging/media/editors/tinymce/js/tinymce.js#L106-L115

```javascript
jQuery(document).on('subform-row-sort', function (event, row) {
    Joomla.JoomlaTinyMCE.setupEditors(row);
});
```

but, the existing editor must first be removed before it is re-initialised, so adding a few lines to do this in the same file :

See - https://github.com/joomla/joomla-cms/blob/staging/media/editors/tinymce/js/tinymce.js#L59-L63

``` javascript
if (element) {
    // We already have the Target, so reset the selector and assign given element as target
    options.selector = null;
    options.target = element;

    var editor = tinymce.get(element.id);

    if (editor) {
        editor.remove();
        delete Joomla.editors.instances[element.id];
    }
}
```

The second part is for the core Tinymce only, but JCE will have it's own fix for this in the next update. The sortable event is required for all editors.

I will try and put this into a PR this coming week.

**Comment by pixxelfactory** [comment]
I can still confirm this bug on the current 3.9.19 release and tried to apply the above fix manually.
However it does not work...
Any way to fix this in the current release?

**Comment by Fedik** [comment]
It is Browser behavior,
Explanation  https://stackoverflow.com/a/2542043/1000711

In short: browser will clear iframe content when DOM changes,

@ryandemmer idea should work

**Comment by Fedik** [comment]
Please test #34808 Joomla 3 fix.

However it still an issue for Joomla 4, so please keep it open for now.

**Comment by richard67** [comment]
@apophis1982 Could you test PR #34808 ? Thanks in advance.

**Comment by richard67** [comment]
@pixxelfactory Could you test PR #34808 ? Thanks in advance.

**Comment by richard67** [comment]
> However it still an issue for Joomla 4, so please keep it open for now.

@Fedik Will you also make a fix for J4?

**Comment by Fedik** [comment]
I have planed, it need more changes there, 
it will be not quick fix, maybe another 2 year ;)

**Comment by richard67** [comment]
Just in time for 4.0 final then ;-)

**Comment by wilsonge** [comment]
Fixed with #34808

**Comment by wilsonge** [comment]
OK Just realised the comment about keeping it open for J4 🤦

**Comment by Fedik** [comment]
The fix for Joomla 4, please test #36345
